### PR TITLE
testing: fix failing test on go 1.15.3

### LINKED ISSUE #96210: Failing test on Go 1.15.3
**Which test(s) are failing**:
`vendor/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go` 

**Since when has it been failing**:
since upgrading golang: `go version go1.15.3 linux/amd64`

**Reason for failure**:
Since Go 1.15,  type casting from int to string using `string(int)` is not allowed, 
output: 
```
# k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create
vendor/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go:75:9: conversion from int to string yields a string of one rune, not a string of digits (did you mean fmt.Sprint(x)?)
FAIL    k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create [build failed]
FAIL
make: *** [Makefile:185: test] Error 1
```

**Anything else we need to know**:
related links: 
golang/go#3939
golang/go#32479
https://blog.golang.org/go1.15-proposals

### PR DESCRIPTION
**Pull request by obeyda**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:
This fixes a failing test in: `staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go:TestCreateRoleBinding`

On go version 1.15.3, casting int to string using `string(int)` causes
the test to fail with this error:

> k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create
vendor/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go:75:9:    conversion from int to string yields a string of one rune, not a    string of digits (did you mean fmt.Sprint(x)?)
FAIL    k8s.io/kubernetes/vendor/k8s.io/kubectl/pkg/cmd/create    [build failed]
FAIL
make: *** [Makefile:185: test] Error 1

**Which issue(s) this PR fixes**:

Fixes #96210

**Special notes for your reviewer**:
See the linked issue's description for more info.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go [hunk 1]
```diff
@@ -72,7 +73,7 @@ func TestCreateRoleBinding(t *testing.T) {
 	}
 
 	for i, tc := range tests {
-		t.Run(string(i), func(t *testing.T) {
+		t.Run(fmt.Sprintf("%d", i), func(t *testing.T) {
```
Comment (knight42) [inline staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go:76]: nit:
```suggestion
		t.Run(strconv.Itoa(i), func(t *testing.T) {
```
`strconv.Itoa` is more efficient in this case.

**Review by knight42 — COMMENTED** [review]
/ok-to-test

File: staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go (hunk 1 above)
Comment (obeyda) [reply staging/src/k8s.io/kubectl/pkg/cmd/create/create_rolebinding_test.go:76]: Agreed!, Sprintf will use `reflect` to allocate an extra object.

**Review by SataQiu — COMMENTED** [review]
/lgtm

**Pull request merged**